great_expectations/rule_based_profiler/parameter_builder/regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,152 @@
+import logging
+from typing import Any, Dict, Iterable, List, Optional, Set, Tuple, Union
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.batch import BatchRequest, RuntimeBatchRequest
+from great_expectations.rule_based_profiler.parameter_builder.parameter_builder import (
+    MetricComputationResult,
+    ParameterBuilder,
+)
+from great_expectations.rule_based_profiler.types import (
+    Domain,
+    ParameterContainer,
+    build_parameter_container,
+)
+from great_expectations.validator.validator import Validator
+
+logger = logging.getLogger(__name__)
+
+
+class RegexPatternStringParameterBuilder(ParameterBuilder):
+    """
+    Detects the domain REGEX from a set of candidate REGEX strings by computing the
+    column_values.match_regex_format.unexpected_count metric for each candidate format and returning the format that
+    has the lowest unexpected_count ratio.
+    """
+
+    CANDIDATE_STRINGS: Set[str] = {
+        r"^\d{1}$",
+        r"^\d{2}$",
+        r"^\S{8}-\S{4}-\S{4}-\S{4}-\S{12}$",
+    }
+
+    def __init__(
+        self,
+        name: str,
+        metric_domain_kwargs: Optional[Union[str, dict]] = None,
+        metric_value_kwargs: Optional[Union[str, dict]] = None,
+        threshold: float = 1.0,
+        candidate_strings: Optional[Iterable[str]] = None,
+        data_context: Optional["DataContext"] = None,
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+    ):
+        """
+        Configure this RegexPatternStringParameterBuilder
+        Args:
+            name: the name of this parameter -- this is user-specified parameter name (from configuration);
+            it is not the fully-qualified parameter name; a fully-qualified parameter name must start with "$parameter."
+            and may contain one or more subsequent parts (e.g., "$parameter.<my_param_from_config>.<metric_name>").
+            threshold: the ratio of values that must match a format string for it to be accepted
+            candidate_strings: a list of candidate date format strings that will REPLACE the default
+            additional_candidate_strings: a list of candidate date format strings that will SUPPLEMENT the default
```

Comment:
Is this supposed to be here? If so, should it be reflected in the signature?

Reply:
done